test/sol4/kyberProxyV1.js

```diff
@@ -0,0 +1,1663 @@
+            //     await token.transfer(taker, amountTWei);
+            //     await token.approve(networkProxyV1.address, amountTWei, {from: taker})
+
+            //     let rates = await networkProxyV1.getExpectedRate(srcToken.address, ethAddress, amountTWei);
+            //     console.log(rates.expectedRate.toString(10));
+
+            //     await expectRevert.unspecified(
+            //         networkProxyV1.tradeWithHint(
+            //             srcToken.address,
+            //             amountTWei,
+            //             ethAddress,
+            //             user1,
+            //             maxDestAmt,
+            //             minConversionRate,
+            //             zeroAddress,
+            //             hint,
+            //             {from: taker}
+            //         )
+            //     );
+            // });
+
+            it("should test can't init this contract with empty contracts (address 0) or with non admin.", async function () {
+                let proxyTemp;
+
+                await expectRevert.unspecified(
+                    NetworkProxyV1.new(zeroAddress),
+                )
+
+                proxyTemp = await NetworkProxyV1.new(admin);
+
+                let rxNetworkAddress = await proxyTemp.kyberNetworkContract();
+                Helper.assertEqual(zeroAddress, rxNetworkAddress, "should be zero address");
+
+                await proxyTemp.setKyberNetworkContract(network.address, {from: admin});
+
+                rxNetworkAddress = await proxyTemp.kyberNetworkContract();
+                Helper.assertEqual(network.address, rxNetworkAddress, "should be correct address");
+
+                await expectRevert.unspecified(
+                    proxyTemp.setKyberNetworkContract(zeroAddress, {from: user1})
+                )
+
+                rxNetworkAddress = await proxyTemp.kyberNetworkContract();
+                Helper.assertEqual(network.address, rxNetworkAddress, "should be correct address");
+            });
+
+            it("should set kyberNetwork and test event.", async function () {
+                let tempNetworkAdd = accounts[7];
+                let result = await networkProxyV1.setKyberNetworkContract(tempNetworkAdd, {from: admin});
+
+                expectEvent(result, 'KyberNetworkSet', {
+                    newNetworkContract: tempNetworkAdd,
+                    oldNetworkContract: network.address,
+                });
+
+                result = await networkProxyV1.setKyberNetworkContract(network.address, {from: admin});
+
+                expectEvent(result, 'KyberNetworkSet', {
+                    newNetworkContract: network.address,
+                    oldNetworkContract: tempNetworkAdd,
+                });
+            });
+        });
+    });
+
+//     it("verify trade is reverted when malicious reserve tries recursive call = tries to call kyber trade function.", async function () {
+
```

Comment:
this will be added later?
still in work?

Reply:
Yeah there are tests with malicious network, but will need to create some mocks for new Network contract